libs/island-ui/core/src/lib/DialogPrompt/DialogPrompt.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  forwardRef,
+  ReactElement,
+  Ref,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react'
+import { FocusableBox } from '../FocusableBox/FocusableBox'
+import { Button } from '../Button/Button'
+import { Box } from '../Box/Box'
+import { Text } from '../Text/Text'
+import { Icon } from '../IconRC/Icon'
+
+import { GridContainer } from '../Grid/GridContainer/GridContainer'
+import { GridRow } from '../Grid/GridRow/GridRow'
+import { GridColumn } from '../Grid/GridColumn/GridColumn'
+import { ModalBase } from '../ModalBase/ModalBase'
+
+import * as styles from './DialogPrompt.treat'
+
+interface DialogPromptProps {
+  title: string
+  description?: string
+  ariaLabel: string
+  baseId: string
+  disclosureElement?: ReactElement
+  onConfirm?: () => void
+  onCancel?: () => void
+  buttonTextConfirm?: string
+  buttonTextCancel?: string
+}
+
+export const DialogPrompt = ({
+  title,
+  description,
+  ariaLabel,
+  baseId,
+  disclosureElement,
+  onConfirm,
+  onCancel,
+  buttonTextCancel = 'Cancel',
+  buttonTextConfirm = 'Confirm',
```

Comment:
not sure about using defaults here, maybe we could use this to conditionally show the buttons, maybe some cases only have one button or no buttons

Reply:
I agree, we should be able to hide the buttons if needed. But I like having default values 😅

Reply:
if we use defaults, wouldn't we need to make it translate-able?

Reply:
Could point, and we don't want to handle translation inside the island-ui, probably good to stick to what you said above then 👍

Reply:
Addressed in https://github.com/island-is/island.is/pull/1474/commits/97880b29409df6fc36c69d4fdd0e8725a71c4a39